### DISCUSSION TITLE
6장 배포시 이슈에 대한 질문이 있습니다

### DISCUSSION THREAD

**Discussion by lutae2000** [fsoftwareengineer/todo-application#65] [질문과 답변] [answered]
안녕하세요 작가님
마침 프론트엔드 & 백엔드 서비스를 나눠서 개발하는 서적을 찾고있던중 마침 좋은책을 집필 해주셔서 감사히 잘 보고 있습니다!
엘라스틱 빈 톡을 이용하여 배포시 문제가 있습니다만 도움을 주시면 감사하겠습니다

![image](https://user-images.githubusercontent.com/40047335/150672343-456dfd88-b7d9-4729-a6da-1f743dedb243.png)
![image](https://user-images.githubusercontent.com/40047335/150672398-211754a9-fcbd-4dc5-9ecb-0f96deb857e4.png)
![image](https://user-images.githubusercontent.com/40047335/150672411-1aff59e5-ac04-426d-920a-fe9363a76fe0.png)

EC2 인스턴스내에 들어가서 빌드된 파일을 직접 실행해보니 계속 개발환경(8080 포트)으로 실행되는거 같습니다
=> 프로덕션으로 정상 기동시 application-prod.yaml 에 있는 server 포트 5000으로 기동되어야 정상인거 같습니다만....

* 서비스가 application-prod.yaml로 실행이 안됨

문제 해결을 위해 해본 부분
- elastic beanTalk에서 앱서버 재기동
- eb 프로덕션 명령어 재입력
- EC2 인스턴시 직접 들어가서 로그 확인
-텔넷으로 RDS와 통신이 정상적인것까지 확인 완료
![image](https://user-images.githubusercontent.com/40047335/150671990-9e8efb50-d51b-4d77-a2a5-cd7f39a6a339.png)

책에 기재된 프로덕션 환경으로 변경하는 명령어 입력했으나 정상 적용이 안되는것으로 예상됩니다
```
$ eb setenv SPRING_PROFILE_ACTIVE=prod
```

프로세스는 올라갔으나 정상적이지 않아 http://[host]/ 경로 접근시 정상적이지 않은것으로 인식하여 ALB에서도 오류 발생
![image](https://user-images.githubusercontent.com/40047335/150672061-846ec2f6-4b8f-49b4-9054-9b2297df0f85.png)

elasticBeanTalk에서 업로드하는 파일 경로 /var/app/current
*아래 스크린샷은 EC2인스턴스에 직접 연결하여 파일 실행시 출력된 로그입니다

![image](https://user-images.githubusercontent.com/40047335/150671630-47cf9ea7-f586-4f3e-b27e-ec25005dc222.png)

**Comment by fsoftwareengineer** [comment, marked as answer]
스프링 로그 상단에 no active profile이라 쓰인것으로 보아 프로파일셋업이 제대로 안된것 같습니다.

https://aws.amazon.com/ko/premiumsupport/knowledge-center/elastic-beanstalk-pass-variables/?nc1=h_ls

AWS콘솔에서 환경변수가 설정되었는지 확인해주세요. 만약 설정되었다면 로컬에서 프로파일을 prod로 실행해보고 프로파일이 제대로 선택되는지 확인해주세요.

**Reply by lutae2000** [reply]
$ eb setenv SPRING_PROFILES_ACTIVE=prod

PROFILES <= 오타가 있었습니다... 답변 감사합니다
